packages/crypto/src/models/wallet.ts

```diff
@@ -123,18 +123,16 @@ export class Wallet {
      * Remove block data from this wallet.
      */
     public revertBlock(block: IBlockData): boolean {
-        this.dirty = true;
-
         if (
             block.generatorPublicKey === this.publicKey ||
             crypto.getAddress(block.generatorPublicKey) === this.address
         ) {
+            this.dirty = true;
             this.balance = this.balance.minus(block.reward).minus(block.totalFee);
 
             // update stats
             this.forgedFees = this.forgedFees.minus(block.totalFee);
             this.forgedRewards = this.forgedRewards.minus(block.reward);
-            this.lastBlock = block;
```

Comment:
The logic here is wrong but simply removing this code won't solve the issue. We should probably store the last 2 or N blocks of a delegate and pop the last block and reattach so we know it is the previous block of the one we removed.